tachyon/math/base/semigroups.h

```diff
@@ -179,6 +179,37 @@ class MultiplicativeSemigroup {
     }
   }
 
+  template <uint64_t Power>
+  [[nodiscard]] constexpr auto ConstPow() const {
```

Comment:
Could you make a similar code named `ConstScalarMul()` and remove switch case from both `Pow()` and `ScalarMul()`?

```suggestion
  template <uint32_t Power>
  [[nodiscard]] constexpr auto ConstPow() const {
```

Reply:
Do you mean remove switch case between `Pow()` and `ConstPow()`?